.github/workflows/code-checks.yaml

```diff
@@ -19,6 +19,7 @@ jobs:
       - name: Setup go
         uses: actions/setup-go@v1
         with:
-          go-version: '^1.16.0'
+          go-version: "^1.17.0"
       - run: make generate
       - run: make lint
+      - run: make test
```

Comment:
I would suggest having 2 jobs instead of one.
It will allow `test` and `lint` to run at the same time and it will allow a quick feedback loop.

Something like this:
```yaml
  Lint:
    runs-on: ubuntu-latest
    env:
      GOPATH: /home/runner/work/open-feature/flagd
    steps:
      - name: Checkout repository
        uses: actions/checkout@v2
      - name: Setup go
        uses: actions/setup-go@v1
        with:
          go-version: "^1.17.0"
      - run: make generate
      - run: make lint

  Test:
    runs-on: ubuntu-latest
    env:
      GOPATH: /home/runner/work/open-feature/flagd
    steps:
      - name: Checkout repository
        uses: actions/checkout@v2
      - name: Setup go
        uses: actions/setup-go@v1
        with:
          go-version: "^1.17.0"
      - run: make generate
      - run: make test
```

Reply:
Great idea!